simulation/__main__.py

```diff
@@ -121,6 +118,13 @@ def check_string_is_isoformat_datetime(string) -> str:
         type=convert_to_datetime,
     )
 
+    parser.add_argument(
+        "--air-source-heat-pump-price-discount-date",
+        action="append",
+        type=map_string_to_datetime_float_tuple,
+        help="A factor by which heat pump prices will fall by a specified date, provided in format 'YYYY-MM-DD:price_discount' (e.g. '2023-01-01:0.3')",
```

Comment:
Use [`metavar`](https://docs.python.org/3/library/argparse.html#metavar) for things like this.

```suggestion
        help="A factor by which heat pump prices will fall by a specified date (e.g. '2023-01-01:0.3')",
        metavar="YYYY-MM-DD:price_discount"
```